mne_bids/write.py

```diff
@@ -413,8 +413,26 @@ def _scans_tsv(raw, raw_fname, fname, overwrite=False, verbose=True):
         # for MNE >= v0.20
         acq_time = meas_date.strftime('%Y-%m-%dT%H:%M:%S.%fZ')
 
-    data = OrderedDict([('filename', ['%s' % raw_fname.replace(os.sep, '/')]),
-                        ('acq_time', [acq_time])])
+    # for fif files check whether raw file is likely to be split
+    raw_fnames = [raw_fname]
+    if raw_fname.endswith('.fif'):
+        # check whether fif files were split when saved
+        # use the files in the target directory what should be written
+        # to scans.tsv
+        datatype, basename = raw_fname.split(os.sep)
+        raw_dir = op.join(op.dirname(fname), datatype)
+        raw_files = [f for f in os.listdir(raw_dir) if f.endswith('.fif')]
+        if basename not in raw_files:
+            raw_fnames = []
+            for raw_f in raw_files:
+                if basename.split('_meg.fif')[0] in raw_f:
+                    raw_fnames.append(op.join(datatype, raw_f))
+            raw_fnames.sort()
```

Comment:
I think this is slightly un-robust.

1. in line 425 you check that the `basename` (e.g., `sub-01_task-test_meg.fif`) is in the list of files.
1. if it is ... all is well, it was not split (no `_split-<idx>` was appended), so we can proceed
1. if it is NOT (the interesting part), you proceed with more checks

See this line: `if basename.split('_meg.fif')[0] in raw_f`

The intention is to use e.g. `sub-01_task-test_meg.fif` to catch files like `sub-01_task-test_split-01_meg.fif`, and `sub-01_task-test_split-02_meg.fif` ... and then put these into a list, order them, and write.

But wouldn't the current code also catch something like `sub-01_task-test_acq-MyParam_meg.fif`?

Reply:
Okay, maybe I don't understand what forms `raw_fname` can take then. I assumed that all the BIDS fields that are part of that recording (sub, task and acq in your example) would be part of `raw_fname` (the only thing missing being the split parameter). So that, the filename on disk would be the same as `raw_fname`, independent of which BIDS fields are included. But you are suggesting that `raw_fname` will not necessarily include all fields (e.g. `acq`)?

Reply:
in other words, I would think line 425 only returns `False` if the files was split and `True` otherwise.

Reply:
> But you are suggesting that raw_fname will not necessarily include all fields (e.g. acq)?

I may be wrong, but couldn't you first write a data file to BIDS without an `acq` entity, ... and then a second one WITH that `acq` entity?

Reply:
Ah, I see. In case there are raw files in the directory that have different number of field, but are the same otherwise? Lines 427-430 would then write twice the file name without `acq` to `scans.tsv`, once because it is found in the file name without `acq` (as it should) and once in the filename with `acq` (because line 428 is too general). Yeah, that's a problem. So better take the basename, add the split field and then check whether there is an exact match.

Reply:
> So better take the basename, add the split field and then check whether there is an exact match.

yes, I think that'd be best :+1:

Reply:
looks a bit hacky now :)

I tested it with some fif files I had available. I suppose these changes should also be covered by the unit test? Like writing one (or more) extra files to that directory and see whether, the correct files will be listed in scans.tsv. Or would that be overkill?

Reply:
> looks a bit hacky now :)

but probably works better! :)

But one more comment: you should use the `.split()` method on `"split"`, but on `"_split-"`, because the former may also catch something like `sub-01_task-Mindsplitting_meg.fif` ("Mindsplitting is just an example task here).

`_split-` is guaranteed to match only the split entity.

>  I suppose these changes should also be covered by the unit test? Like writing one (or more) extra files to that directory and see whether, the correct files will be listed in scans.tsv. Or would that be overkill?

If you can manage to cover this case in a couple of lines, that'd be awesome